### ISSUE TITLE
Communication channels

### ISSUE THREAD

**Issue by tarcieri** [The-DevX-Initiative/RCIG_Coordination_Repo#1]
This is a tracking issue to discuss what communication channels we should use going forward.

Open questions:

1. Chat: should we set up a Discord/Matrix/Zulip for this group? Should we use an existing channel e.g. #crypto-and-security on the Rust Programming Language Community Discord?
    a. chat topics:
        - elliptic curve cryptography
        - password hashing / PAKEs
        - post-quantum cryptography
        - symmetric cryptography
        - zero-knowledge proofs
        - traits/interop
        - secret independent coding
        - target specific code/issues
        - governance and vision
        - others? (please leave a comment)
2. Email/forums: should we set up a mailing list or something like Discourse?
3. Meetings: should we have regular meetings? If so, what kinds?
    a. synchronous video calls?
    b. async meetings via chat?

Edit: we decided to go with Zulip! Here is an invite link to join:

https://rcig.zulipchat.com/join/kr6lqdw4x554skbscmjttvhg/

I've also enabled the Discussions feature for this repo:

https://github.com/The-DevX-Initiative/RCIG_Coordination_Repo/discussions/7

**Comment by richsalz** [comment]
yeah for zulip!  but it needs a host.

**Comment by cronokirby** [comment]
I think using the #crypto-and-security channel might get a bit cluttered with different topics.

For chat I'm more familiar with (and biased towards) Discord, but other Rust and OSS projects seem to use Zulip to great effect.

I think having a forum is perhaps a bit redundant if we have a chat group. Personally, I'm more partial towards chat groups anyways.

**Comment by tuxxy** [comment]
Would much prefer discord over Zulip personally. All my chats are in
Discord these days. :)

**Comment by tarcieri** [comment]
Expanding on chat, I added a sub-topic (1a) to the toplevel description to discuss chat topics. Please let me know which ones you're interested in.

**Comment by riking** [comment]
> Email/forums: should we set up a mailing list or something like Discourse?

It really feels like there should be some way to swing per-interest-group spaces on a shared forum.

Maybe private categories + "anyone can join" groups on URLO would do it?

**Comment by franziskuskiefer** [comment]
>  a. chat topics:

I'd extend the topics with more general things like
* traits/interop
* secret independent coding
* target specific code/issues
* governance
* vision (doc)

**Comment by spitters** [comment]
Zulip seems to work well, both for hacspec, and sometimes in the rust formal verification stream.

**Comment by tarcieri** [comment]
Zulip vs Discord seems like a tough call. FWIW I use them both.

I think Discord has better UX overall, but Zulip streams are really nice for organizing conversations into forum-like threads, at least if people actually understand/use them (which has also been a problem).

I suppose I'm weakly in favor of a Discord to get started. I think Zulip is nice when you have a lot of discussion that needs to be organized/topic focused, but until there's a lot of ongoing discussion perhaps streams are a bit of a premature optimization.

**Comment by dfaranha** [comment]
+1 to Zulip, as I find Discord a bit too distracting

**Comment by dconnolly** [comment]
I would vote in favor of Discord if only because we can also use it as the AV meeting platform for sync discussions as needed (afaik Zulip doesn't have video/audio calling?), so that we wouldn't have to do a round of Zoom vs Meet vs Jitter, etc.

**Comment by tarcieri** [comment]
If I counted correctly that's 5 in favor of Discord, 4 in favor of Zulip so far. Pretty evenly split 😬

I think the AV issue is a good one in terms of having one platform for both and having them integrated. It looks like it supports up to 50 people?

**Comment by richsalz** [comment]
can i keep a browser window open to watch discord chats?  sure, sigh, what's yet another messaging platform (i know, anyone in this group could say that :)

**Comment by tarcieri** [comment]
Yeah, it has a web UI with much of the functionality of the native app

**Comment by rozbb** [comment]
fwiw I'm pro-Zulip. I've had enough bad experiences with Discord audio and video that I don't view that feature as a plus. I think the "premature optimization" argument re threading is fair, but I think the argument is weaker when you note that this choice of communication mechanism is almost certainly a permanent one (migrating is probably more painful than whatever pains we encounter).

I think Zulip's support for threads and LaTeX make it a very good choice for a crypto working group. The primary downsides imo are that this doesn't solve synchronous communication, and it's probably less likely that a newcomer will have a Zulip account over a Discord account. For synchronous comms, I think we can discuss that separately. For newcomers, I think the importance of this depends on how much membership influx we expect.

All this said, I've got accounts on both platforms, so pick away :)

**Comment by spitters** [comment]
Zulip was also used quite effectively for the rust verification workshop. 
It the common platform for most people in verification. I currently have 8 of them open. It would be harder for me to keep track of discord.

**Comment by newpavlov** [comment]
+1 for Zulip. :)

**Comment by tarcieri** [comment]
For Zulip fans, please check out the existing one for the https://github.com/RustCrypto project:

https://rustcrypto.zulipchat.com/

**Comment by spitters** [comment]
@tarcieri What's the reason for having a separate zulip for rustcrypto. I find it easier to follow streams in the general rust zulip. It would also increase the visibility of the IG, I think.

**Comment by tarcieri** [comment]
@spitters pretty much the same reasons something like Hacspec has its own Zulip: because it isn't affiliated with the Rust Project, and because there are enough topics to warrant breaking down the conversation into separate streams per algorithm category

**Comment by ineiti** [comment]
+1 for zulip, too, but only because I wanted to try it out for a long time already and never came around to do so ;)

**Comment by franziskuskiefer** [comment]
@tarcieri I think we just need to go for something. As long as there's no official governance how about you pick the chat app?

> Email/forums: should we set up a mailing list or something like Discourse?

We could open the discussion section here on GH for an easily accessible forum.

**Comment by tarcieri** [comment]
Okay, it seems there are enough people in favor of Zulip at this point we can go ahead and use that.

Here's an invite link:

https://rcig.zulipchat.com/join/rrqhlitutey3ilhjdzralyzs/

**Comment by spitters** [comment]
Somehow the link does not work for me.

**Comment by tarcieri** [comment]
@spitters try this one: https://rcig.zulipchat.com/join/ex4wudpgdapacdvakexshuio/

**Comment by tarcieri** [comment]
For people who would prefer a more forum-like experience, I've enabled the Discussions feature for this repo:

https://github.com/The-DevX-Initiative/RCIG_Coordination_Repo/discussions/7

**Comment by tarcieri** [comment]
Going to briefly reopen this.

On the most recent call it sounded like there was some decent support for a mailing list.

I'm on a number of cryptography-related mailing lists and at least within this particular subject area it seems like something people do use. Should we make one?

**Comment by richsalz** [comment]
I would have preferred mailing lists, but we already have two zulip channels and I would not like to fracture the conversation.

**Comment by sebastianganson** [comment]
I completely agree about minimizing fractured conversation, although am old school and find my brain works better with organizing emails :-|

I’d be interested to hear more about recommended cryptography mailing lists. I currently belong to a couple related to PQC and HE, but am always on the lookout for other ones. 

Best Regards,
Sebastian

**Comment by rozbb** [comment]
fwiw Zulip has some amount of mailing list integration

https://zulip.com/help/message-a-stream-by-email
